llvm/projects/evm-tests/dynamic_stackalloc.cpp

```diff
@@ -0,0 +1,55 @@
+
+struct Array {
+  __attribute__ ((noinline)) Array(unsigned char* p, unsigned sz): buf(p), size(sz) {}
+
+  void __attribute__ ((noinline)) fill(unsigned char addend) {
+    for (unsigned i = 0; i < size; i++) {
+      buf[i] = i + addend;
+    }
+  }
+
+  unsigned char *buf = nullptr;
+  unsigned size = 0;
+};
+
+int __attribute__ ((noinline)) nested_alloc(Array& a1, Array& a2, unsigned i) {
+  auto size = a1.size;
+  unsigned char buf[size];
```

Comment:
Are dynamically sized arrays on the stack a standard C++ feature? From what I remember, it is present in C99. 
The code that I worked on usually utilized `alloca()` for this purpose.

Not saying that we should change it, just interesting.

Reply:
`alloca` is the same as dynamic stack arrays, and you're right, they both is not in C++ standard, they are in C99. But it is only the option how we can implement dynamically sized data for now.